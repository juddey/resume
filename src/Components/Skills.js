import React from 'react'
import LinearProgress from '@material-ui/core/LinearProgress'
import Typography from '@material-ui/core/Typography'
import Grid from '@material-ui/core/Grid'
import { withStyles } from '@material-ui/core'
import data from '../data/Skill'

// TODO: Can we put this object in its own file?
const styles = theme => ({
  container: {
    display: 'flex',
    alignItems: 'center'
  },
  colorPrimary: {
    backgroundColor: '#F8F8F8'
  },
  barColorPrimary: {
    backgroundColor: '#012C4C'
  }
})

function Skills (props) {
  const { classes } = props
  return (
    <div>
      {data.map((x, i) => (
        <div key={i} className={classes.container}>
          <Grid item xs={3} md={2}>
            <Typography>{x.label}</Typography>
          </Grid>
          <Grid item xs={9} md={10}>
            <LinearProgress
              classes={{
                barColorPrimary: classes.barColorPrimary,
                colorPrimary: classes.colorPrimary
              }}
              variant='determinate'
              value={x.level}
            />
          </Grid>
        </div>
      ))}
    </div>
  )
}

export default withStyles(styles)(Skills)
