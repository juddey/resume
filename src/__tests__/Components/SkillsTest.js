import React from 'react'
import {
  createComponentWithIntl,
  shallowWithIntl
} from '../../../src/Support/Utils'
import Skills from '../../Components/Skills'

const div = document.createElement('div')
const wrapper = shallowWithIntl(<Skills />)

test('component exists', () => {
  expect(wrapper.length).toBe(1)
})

test('component structure', () => {
  expect(wrapper.name()).toEqual('Skills')
  expect(wrapper.children().length).toEqual(0)
})

test('renders to snapshot', () => {
  const tree = createComponentWithIntl(<Skills />, div).toJSON()
  expect(tree).toMatchSnapshot()
})
