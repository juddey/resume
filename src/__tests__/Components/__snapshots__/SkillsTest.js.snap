// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to snapshot 1`] = `
<div>
  <div
    className="Skills-container-1"
  >
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-3-33 MuiGrid-grid-md-2-58"
    >
      <p
        className="MuiTypography-root-95 MuiTypography-body1-104"
      >
        React Native
      </p>
    </div>
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-9-39 MuiGrid-grid-md-10-66"
    >
      <div
        aria-valuenow={85}
        className="MuiLinearProgress-root-119 MuiLinearProgress-colorPrimary-120 Skills-colorPrimary-2"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar-127 MuiLinearProgress-barColorPrimary-128 Skills-barColorPrimary-3 MuiLinearProgress-bar1Determinate-132"
          style={
            Object {
              "transform": "scaleX(0.85)",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="Skills-container-1"
  >
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-3-33 MuiGrid-grid-md-2-58"
    >
      <p
        className="MuiTypography-root-95 MuiTypography-body1-104"
      >
        React
      </p>
    </div>
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-9-39 MuiGrid-grid-md-10-66"
    >
      <div
        aria-valuenow={70}
        className="MuiLinearProgress-root-119 MuiLinearProgress-colorPrimary-120 Skills-colorPrimary-2"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar-127 MuiLinearProgress-barColorPrimary-128 Skills-barColorPrimary-3 MuiLinearProgress-bar1Determinate-132"
          style={
            Object {
              "transform": "scaleX(0.7)",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="Skills-container-1"
  >
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-3-33 MuiGrid-grid-md-2-58"
    >
      <p
        className="MuiTypography-root-95 MuiTypography-body1-104"
      >
        HTML
      </p>
    </div>
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-9-39 MuiGrid-grid-md-10-66"
    >
      <div
        aria-valuenow={45}
        className="MuiLinearProgress-root-119 MuiLinearProgress-colorPrimary-120 Skills-colorPrimary-2"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar-127 MuiLinearProgress-barColorPrimary-128 Skills-barColorPrimary-3 MuiLinearProgress-bar1Determinate-132"
          style={
            Object {
              "transform": "scaleX(0.45)",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="Skills-container-1"
  >
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-3-33 MuiGrid-grid-md-2-58"
    >
      <p
        className="MuiTypography-root-95 MuiTypography-body1-104"
      >
        JavaScript
      </p>
    </div>
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-9-39 MuiGrid-grid-md-10-66"
    >
      <div
        aria-valuenow={65}
        className="MuiLinearProgress-root-119 MuiLinearProgress-colorPrimary-120 Skills-colorPrimary-2"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar-127 MuiLinearProgress-barColorPrimary-128 Skills-barColorPrimary-3 MuiLinearProgress-bar1Determinate-132"
          style={
            Object {
              "transform": "scaleX(0.65)",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="Skills-container-1"
  >
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-3-33 MuiGrid-grid-md-2-58"
    >
      <p
        className="MuiTypography-root-95 MuiTypography-body1-104"
      >
        Ruby
      </p>
    </div>
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-9-39 MuiGrid-grid-md-10-66"
    >
      <div
        aria-valuenow={80}
        className="MuiLinearProgress-root-119 MuiLinearProgress-colorPrimary-120 Skills-colorPrimary-2"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar-127 MuiLinearProgress-barColorPrimary-128 Skills-barColorPrimary-3 MuiLinearProgress-bar1Determinate-132"
          style={
            Object {
              "transform": "scaleX(0.8)",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="Skills-container-1"
  >
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-3-33 MuiGrid-grid-md-2-58"
    >
      <p
        className="MuiTypography-root-95 MuiTypography-body1-104"
      >
        Elixir
      </p>
    </div>
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-9-39 MuiGrid-grid-md-10-66"
    >
      <div
        aria-valuenow={70}
        className="MuiLinearProgress-root-119 MuiLinearProgress-colorPrimary-120 Skills-colorPrimary-2"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar-127 MuiLinearProgress-barColorPrimary-128 Skills-barColorPrimary-3 MuiLinearProgress-bar1Determinate-132"
          style={
            Object {
              "transform": "scaleX(0.7)",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
